### PR TITLE
Try bail instead of mocha-steps

### PR DESCRIPTION
**Pull request by scinos**
#### Background

To understand the goal of this PR it is good to have some background on how Magellan and Mocha work:

* When Magellan starts it will run find all `describe` blocks tagged with `@parallel`. Then, it wil send each `@parallel` block to a separate Mocha process. This means that form Mocha's point of view, there is one and only one suite being run.

* `mocha-steps` implements `step` as a replacement for `it`. The main benefit is that when a `step` fails, it will skip the rest of `step` tests in the same `describe` block.

* By default, whenan `it` block fails, other `it` blocks in the same `describe` block will continue running. This behaviour can be changed with the `--bail` flag, so when an `it` fails, the whole test is aborted.

With that in mind, I think `mocha-steps` is redundant for us: as Mocha is running only one suite tagged with `@parallel`, aborting it when a test fails (i.e. use `--bail`) is virtually the same than using `step`.

There is a caveat to consider: nested describe blocks. Imagine there is a `@parallel` block that has two `describe` blocks inside. When using `it`+`--bail`, it will not run the second `describe` block if the first one fails. When using `step` it will still run the second `describe` block. 

With the current behaviour both `describe` blocks may fail. In general, there may be more than one failing tests per Mocha run. This makes capturing screenshots, video recordings and logs from tests more complicated, as now you need to make sure you don't overwrite files from previous tests in the same Mocha process. 

#### Changes proposed in this Pull Request

* Drop `mocha-steps` and use `it` + `--bail` instead.

#### Testing instructions

* Check e2e tests still pass.

### CODE REVIEW

**Comment by worldomonation** [comment]
Checked out the E2E Desktop run and found that one suite, `[WPCOM] Site Editor (desktop)` failed in the first try, and passed on second try. So that's good.

Also pulled the branch locally then introduced a failure at [Line 41 of reader spec](https://github.com/Automattic/wp-calypso/blob/abae1ba19bb757b00e2907b7af9398639a39efcf/test/e2e/specs/wp-reader-spec.js#L41) and ran the spec with `mocha --bail` option set. The test stopped executing after encountering the error. Good.

I have to evaluate what this means for my Playwright work however. I will try to either rebase my work onto this branch or wait until this branch is merged into trunk before continuing.

File: test/e2e/.mocharc.yml
Comment (worldomonation) [inline test/e2e/.mocharc.yml:7]: I think it would make sense to also place the `bail: true` option at the mocha config level, as I had forgotten to set that flag in the CLI when testing locally and was baffled why rest of the steps attempted to run.

**Review by worldomonation — APPROVED** [review]

**Comment by noahtallen** [comment]
A couple questions:
- Is there ever a case when "it" blocks don't depend on the others? (like, do we ever want the "normal" behavior?)
- Would devs find it confusing for "it" to work like this here when it doesn't in the other test suites (like unit)?

**Comment by scinos** [comment]
> Is there ever a case when "it" blocks don't depend on the others? (like, do we ever want the "normal" behavior?)

I'm not sure. I guess it won't be very common for e2e tests. If we do, it can opted-out from inside the test using `this.bail()` (https://mochajs.org/api/mocha#bail).

> Would devs find it confusing for "it" to work like this here when it doesn't in the other test suites (like unit)?

Good point. I don't know :) Personally, I would find `it`+`--bail` less confusing than mocha-steps, because it is not clear why I can't use steps in unit tests, for example. But it is subjective, of course.

**Pull request merged**
